test/tests/functional/operation_example_tests.js

```diff
@@ -1,873 +1,850 @@
-      _server.command("system.$cmd", {ismaster: true}, function(err, result) {
-        test.equal(null, err)
-        server.destroy();
-        test.done();
+  });
+
+  /**
+   * Correctly execute ismaster command on the ReplSet using the cursor
+   *
+   * @example-class ReplSet
+   * @example-method command
+   * @ignore
+   */
+  it('command using ReplSet instance',  {
+    metadata: {
+      requires: {
+        topology: 'replicaset'
+      }
+    },
+
+    test: function(done) {
+      var ReplSet = this.configuration.require.ReplSet,
+          ReadPreference = this.configuration.require.ReadPreference;
+
+      var config = [{
+        host: this.configuration.host,
+        port: this.configuration.port
+      }];
+
+      var options = {
+        setName: this.configuration.setName
+      };
+
+      // Attempt to connect
+      var server = new ReplSet(config, options);
+
+      // LINE var Server = require('mongodb-core').Server,
+      // LINE   test = require('assert');
+      // LINE var server = new ReplSet([{host: 'localhost', port:31000}], {setName:'rs'});
+      // REMOVE-LINE done();
+      // BEGIN
+      // Add event listeners
+      server.on('connect', function(_server) {
+        // Execute the command
+        _server.command('system.$cmd', {ismaster: true}, function(err, result) {
+          expect(err).to.be.null;
+          server.destroy();
+          done();
+        });
       });
-    });
 
-    // Start connection
-    server.connect();
-    // END
-  }
-}
+      // Start connection
+      server.connect();
+      // END
+    }
+  });
+});
 
 /**************************************************************************
  *
  * MONGOS TESTS
  *
  *************************************************************************/
 
-/**
- * Correctly insert a document using the Mongos insert method
- *
- * @example-class Mongos
- * @example-method insert
- * @ignore
- */
-exports['Example of simple insert into db using Mongos'] = {
-  metadata: {
-    requires: {
-      topology: "mongos"
-    }
-  },
-
-  test: function(configuration, test) {
-    var Mongos = configuration.require.Mongos;
-
-    // Attempt to connect
-    var server = new Mongos([{
-        host: configuration.host
-      , port: configuration.port
-    }]);
-
-    // LINE var Mongos = configuration.require.Mongos,
-    // LINE   test = require('assert');
-    // LINE var server = new Mongos([{host: 'localhost', port:50000}]);
-    // REMOVE-LINE test.done();
-    // BEGIN
-    // Add event listeners
-    server.on('connect', function(_server) {
-      // Execute the insert
-      _server.insert('integration_tests.inserts_example_mongos_1', [{a:1}], {
-        writeConcern: {w:1}, ordered:true
-      }, function(err, results) {
-        test.equal(null, err);
-        test.equal(1, results.result.n);
-
-        _server.destroy();
-        test.done();
-      });
-    });
+describe.skip('Mongos operation example tests', function() {
```

Comment:
not sure what to do here, since there is no `mongos` topology to setup, and running the tests under the `sharded` topology causes them to error